packages/fetchai/connections/oef/connection.py

```diff
@@ -503,17 +510,20 @@ def on_dialogue_error(
         """
         assert self.in_queue is not None
         assert self.loop is not None
-        msg = OEFMessage(
-            type=OEFMessage.Type.DIALOGUE_ERROR,
-            id=answer_id,
-            dialogue_id=dialogue_id,
-            origin=origin,
+        msg = DefaultMessage(
```

Comment:
Note, @marcofavorito it makes little sense to use the OEF protocol for notifying of dialogue errors as these would be caused by protocols other than the OEF protocol. It makes sense to use  the default protocol instead so that the error handler can kick in. Otherwise, each agent needs the oef protocol just for error handling, which will confuse people.

Reply:
But can't a protocol have its own error messages and error handling, like in the OEF protocol?

Reply:
The dialogue error would result from a different protocol (i.e. not oef protocol) and should therefore error out on base protocol (default protocol). Ex: I send a message in fipa via oef which causes a dialogue error on oef node..it's now not ok for the oef node to send me an error in oef protocol as I might not have that protocol, instead it should use default protocol